boa_interner/src/sym.rs

```diff
@@ -0,0 +1,149 @@
+
+    /// Symbol for the `"eval"` string.
+    pub const EVAL: Self = unsafe { Self::new_unchecked(5) };
+
+    /// Symbol for the `"default"` string.
+    pub const DEFAULT: Self = unsafe { Self::new_unchecked(6) };
+
+    /// Symbol for the `"null"` string.
+    pub const NULL: Self = unsafe { Self::new_unchecked(7) };
+
+    /// Symbol for the `"RegExp"` string.
+    pub const REGEXP: Self = unsafe { Self::new_unchecked(8) };
+
+    /// Symbol for the `"get"` string.
+    pub const GET: Self = unsafe { Self::new_unchecked(9) };
+
+    /// Symbol for the `"set"` string.
+    pub const SET: Self = unsafe { Self::new_unchecked(10) };
+
+    /// Symbol for the `"<main>"` string.
+    pub const MAIN: Self = unsafe { Self::new_unchecked(11) };
+
+    /// Symbol for the `"raw"` string.
+    pub const RAW: Self = unsafe { Self::new_unchecked(12) };
+
+    /// Symbol for the `"static"` string.
+    pub const STATIC: Self = unsafe { Self::new_unchecked(13) };
+
+    /// Symbol for the `"prototype"` string.
+    pub const PROTOTYPE: Self = unsafe { Self::new_unchecked(14) };
+
+    /// Symbol for the `"constructor"` string.
+    pub const CONSTRUCTOR: Self = unsafe { Self::new_unchecked(15) };
+
+    /// Symbol for the `"implements"` string.
+    pub const IMPLEMENTS: Self = unsafe { Self::new_unchecked(16) };
+
+    /// Symbol for the `"interface"` string.
+    pub const INTERFACE: Self = unsafe { Self::new_unchecked(17) };
+
+    /// Symbol for the `"let"` string.
+    pub const LET: Self = unsafe { Self::new_unchecked(18) };
+
+    /// Symbol for the `"package"` string.
+    pub const PACKAGE: Self = unsafe { Self::new_unchecked(19) };
+
+    /// Symbol for the `"private"` string.
+    pub const PRIVATE: Self = unsafe { Self::new_unchecked(20) };
+
+    /// Symbol for the `"protected"` string.
+    pub const PROTECTED: Self = unsafe { Self::new_unchecked(21) };
+
+    /// Symbol for the `"public"` string.
+    pub const PUBLIC: Self = unsafe { Self::new_unchecked(22) };
+
+    #[inline]
+    /// Creates a new [`Sym`] from the provided `value`, or returns `None` if `index` is zero.
+    pub(super) fn new(value: usize) -> Option<Self> {
+        NonZeroUsize::new(value).map(|value| Self { value })
+    }
+
+    #[inline]
+    /// Creates a new [`Sym`] from the provided `value`, without checking if `value` is not zero
+    ///
+    /// # Safety
+    ///
+    /// `value` must not be zero.
+    pub(super) const unsafe fn new_unchecked(value: usize) -> Self {
+        Self {
+            value:
+            // SAFETY: The caller must ensure the invariants of the function.
+            unsafe {
+                NonZeroUsize::new_unchecked(value)
+            },
+        }
+    }
+
+    #[inline]
+    /// Returns the internal value of the [`Sym`]
+    pub(super) const fn get(self) -> usize {
+        self.value.get()
+    }
+}
+
+// Safe because `Sym` implements `Copy`.
+unsafe impl Trace for Sym {
+    unsafe_empty_trace!();
+}
```

Comment:
Should this be implemented, or should we force users of `Sym` to use `#[unsafe_ignore_trace]`?

Reply:
Had to do some shenanigans with the implementation of `Object`, but with that `Sym` no longer depends on `gc`